### PR TITLE
Don't save mac metadata/extended attributes for `brew bottle`

### PR DESCRIPTION
**Pull request by ywwry66**
This commit includes `--no-mac-metadata` `--no-acls` and `--no-xattrs` in `default_tar_args` for `brew bottle` command.

Although `default_tar_args` is only active when `--only-json-tab` is not passed, in which case we don't require reproducible bottles, it is nonetheless beneficial to "regularize" tarball creation. In particular, this resolves a sporadic `brew tests
--only=dev-cmd/bottle:20` failure (see
https://github.com/orgs/Homebrew/discussions/4376 and https://github.com/Homebrew/brew/pull/14997).

Furthermore, with `gnu tar`, `--no-acls` and `--no-xattrs` are default flags. As for "mac metadata", although I couldn't find official documentation, this post (https://superuser.com/a/61188) shares some info:
- Resource forks (resource forks have been extended attributes since 10.4)
  - Custom icons set in Finder and the images of Icon\r files
  - Metadata in PSD files
  - Objects stored in scpt files, AppleScript Editor window state, descriptions of scripts
- Information about aliases (aliases stop working if extended attributes are removed)
- Quarantine status or source URLs of files downloaded from the internet
- Spotlight comments
- Encoding of files saved with TextEdit
- Caret position of files opened with TextMate
- Skim notes

None of these is supposed to be in the bottle I believe.

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

File: Library/Homebrew/dev-cmd/bottle.rb [hunk 1]
```diff
@@ -232,8 +232,11 @@ def self.sudo_purge
   end
 
   def self.setup_tar_and_args!(args, mtime)
+    # TODO: Refactor and move to extend/os
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/bottle.rb:235]: Sorry, would really like this to be done in this PR

**Review by MikeMcQuaid — COMMENTED** [review]
Thanks for the PR!

> Furthermore, with gnu tar, --no-acls and --no-xattrs are default flags. 

If they are default: any need to add them here?

File: Library/Homebrew/dev-cmd/bottle.rb (hunk 1 above)
Comment (ywwry66) [reply Library/Homebrew/dev-cmd/bottle.rb:235]: Sure, I will give a try.

> If they are default: any need to add them here?

I added those for "generic OS" that CI tests against. But we can definitely remove that for real life usage.

File: Library/Homebrew/dev-cmd/bottle.rb [hunk 2]
```diff
@@ -231,23 +231,29 @@ def self.sudo_purge
...
-    default_tar_args = ["tar", [].freeze].freeze
-    return default_tar_args unless args.only_json_tab?
+  sig { returns(T::Array[String]) }
+  def self.tar_args
+    [].freeze
+  end
 
-    # Ensure tar is set up for reproducibility.
+  sig { params(mtime: String).returns(T::Array[String]) }
+  def self.gnutar_args(mtime)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/bottle.rb:240]:
```suggestion
  def self.reproducible_gnutar_args(mtime)
```

maybe?

**Review by MikeMcQuaid — COMMENTED** [review]
Great work, almost there!

File: Library/Homebrew/extend/os/linux/dev-cmd/bottle.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# typed: true
+# frozen_string_literal: true
+
+module Homebrew
+  sig { params(args: T.untyped, mtime: String).returns([String, T::Array[String]]) }
+  def self.setup_tar_and_args!(args, mtime)
+    # Without --only-json-tab bottles are never reproducible
+    return ["tar", tar_args].freeze unless args.only_json_tab?
```
[the remark this answers is not in the document]
Comment (ywwry66) [reply Library/Homebrew/extend/os/linux/dev-cmd/bottle.rb:8]: It is possible, but I end up with more lines of code and more complicated logic so I guess it is better to leave as is.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by ywwry66** [comment]
Test failure is irrelevant.

**Review by apainintheneck — APPROVED** [review]

**Pull request merged**